core/src/main/java/io/temporal/samples/hello/HelloAccumulator.java

```diff
@@ -0,0 +1,634 @@
+    * code, network calls, database operations, etc. Those things should be handled by the
+    * Activities.
+    *
+    * @see io.temporal.workflow.WorkflowInterface
+    * @see io.temporal.workflow.WorkflowMethod
+    */
+   @WorkflowInterface
+   public interface AccumulatorWorkflow {
+     /**
+      * This is the method that is executed when the Workflow Execution is started. The Workflow
+      * Execution completes when this method finishes execution.
+      */
+     @WorkflowMethod
+     String accumulateGreetings(
+         String bucketKey, Deque<Greeting> greetings, Set<String> allGreetingsSet);
+ 
+     // Define the workflow sendGreeting signal method. This method is executed when
+     // the workflow receives a greeting signal.
+     @SignalMethod
+     void sendGreeting(Greeting greeting);
+ 
+     // Define the workflow exit signal method. This method is executed when the
+     // workflow receives an exit signal.
+     @SignalMethod
+     void exit();
+   }
+ 
+   /**
+    * This is the Activity Definition's Interface. Activities are building blocks of any Temporal
+    * Workflow and contain any business logic that could perform long running computation, network
+    * calls, etc.
+    *
+    * <p>Annotating Activity Definition methods with @ActivityMethod is optional.
+    *
+    * @see io.temporal.activity.ActivityInterface
+    * @see io.temporal.activity.ActivityMethod
+    */
+   @ActivityInterface
+   public interface GreetingActivities {
+     String composeGreeting(Deque<Greeting> greetings);
+   }
+ 
+   /** Simple activity implementation. */
+   static class GreetingActivitiesImpl implements GreetingActivities {
+ 
+     // here is where we process all of the signals together
+     @Override
+     public String composeGreeting(Deque<Greeting> greetings) {
+       List<String> greetingList =
+           greetings.stream().map(u -> u.greetingText).collect(Collectors.toList());
+       return "Hello (" + greetingList.size() + ") robots: " + greetingList + "!";
+     }
+   }
+ 
+   // Main workflow method
+   public static class AccumulatorWorkflowImpl implements AccumulatorWorkflow {
+ 
+     private final GreetingActivities activities =
+         Workflow.newActivityStub(
+             GreetingActivities.class,
+             ActivityOptions.newBuilder().setStartToCloseTimeout(Duration.ofSeconds(2)).build());
+ 
+     private static final Logger logger = LoggerFactory.getLogger(AccumulatorWorkflowImpl.class);
+     String bucketKey;
+     ArrayDeque<Greeting> greetings;
+     HashSet<String> allGreetingsSet;
+     boolean exitRequested = false;
+     ArrayDeque<Greeting> unprocessedGreetings;
+ 
+     @Override
+     public String accumulateGreetings(
+         String bucketKeyInput, Deque<Greeting> greetingsInput, Set<String> allGreetingsSetInput) {
+       bucketKey = bucketKeyInput;
+       if (unprocessedGreetings == null) unprocessedGreetings = new ArrayDeque<Greeting>();
```

Comment:
nit pick: you don't need this check you can initialize `unprocessedGreetings` on line 195 and avoid this check and the same check in the signal handler

Reply:
You can see use do it in this sample https://github.com/temporalio/samples-java/blob/main/core/src/main/java/io/temporal/samples/moneybatch/AccountTransferWorkflowImpl.java#L43